src/cmdlib.sh

```diff
@@ -93,12 +93,16 @@ depcheck() {
 preflight() {
     depcheck
 
-    if [ "$(stat -f --printf="%T" .)" = "overlayfs" ]; then
-        fatal "$(pwd) must be a volume"
-    fi
+    # When OCP_CUSTOM_BUILDER is set to 1, we expect low-level permissions
```

Comment:
AIUI in 4.x custom builders are somewhat deprecated because the goal was to move builds to be unprivileged.
In particular in 3.x it was really easy to mount the docker socket but that grants total control over the host.

In 4.x see e.g. https://docs.openshift.com/container-platform/4.5/builds/custom-builds-buildah.html

And AIUI these builds happen without any privileges.

I suspect the reason this is working for you is if you were testing it in the privileged.psi, the pod spec for the builder might have automatically used a privileged security context constraint or so?

If we can get this to work in a dev namespace for CentOS CI (xref https://pagure.io/centos-infra/issue/83 ) where we don't have privileges (just KVM requested) then it would apply to the current FCOS pipeline and also align with what we need to do for RHCOS (move off of requesting `privileged: true` because we don't need it).

Reply:
From that doc:
```

Using the custom build strategy requires permissions that normal users do not have by default because it allows the user to execute arbitrary code inside a privileged container running on the cluster. This level of access can be used to compromise the cluster and therefore should be granted only to users who are trusted with administrative privileges on the cluster.
```

HOWEVER, in the next push, there will be a defined `spec.serviceAccount` template. The important bit that gives the access is because our build strategy is "custom" which is a bit different than custom builders; a custom build strategy is different than a custom builder.

Reply:
OK yep if we can control the serviceaccount used for the builder pod, then that sounds good.

Reply:
Before I lift the WIP, I'll run down making sure that this works on 4.x. I'm trying to hunt down access to the new CentOS CI to confirm.